lib/oauthutil/oauthutil.go

```diff
@@ -443,11 +466,20 @@ func (s *authServer) Start() {
 			}
 		} else {
 			fs.Debugf(nil, "No code found on request")
+			httpResponse := "<h1>Failed!</h1>\nNo code found returned by remote server."
+			if s.errorHandler != nil {
+				authError := s.errorHandler(req)
+				errorDesc := fmt.Sprintf("Error: %s\nCode: %s\nDescription: %s\nHelp: %s",
+					authError.Name, authError.Code, authError.Description, authError.HelpURL)
+				httpResponse += "\n<p>" + strings.Replace(errorDesc, "\n", "<br>", -1) + "</p>"
+				err = errors.New(errorDesc)
```

Comment:
I wonder whether we should use an html.Template for this as there is a potential for HTML injection here.

It might be neater if you make a default errorHandler then call it regardless here.  The default error handler can return an AuthError too.

If we have a template which we pass AuthError to then it would be easy to make HelpURL clicky too!

I don't know if you are familiar with go templates, but there is a simple example here which you can look at:

https://github.com/ncw/rclone/blob/2a29f7f6c8780a7d6487559d6d5dd8c26dc9e14f/cmd/serve/http/http.go#L143